File: dp_labs_and_solutions/dp_labs_and_solutions/dp_cpp_labs_and_solutions/null_object/solution/binary_minus.cpp
// binary_minus.cpp

#include "binary.h"
#include "binary_minus.h"
class expression;
#include "expression_visitor.h"

binary_minus::binary_minus(expression * a_operand1, expression * a_operand2)
: binary(a_operand1, a_operand2)
{
}

void binary_minus::accept(expression_visitor & a_expression_visitor)
{
	a_expression_visitor.visit(*this);
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_java_labs_and_solutions/null_object/solution/BinaryMinus.java
// BinaryMinus.java

public class BinaryMinus extends Binary
{
	public BinaryMinus(Expression aOperand1, Expression aOperand2)
	{
		super(aOperand1, aOperand2);
	}

	public void accept(ExpressionVisitor aExpressionVisitor)
	{
		aExpressionVisitor.visit(this);
	}
}
